OData/test/UnitTest/System.Web.OData.Test/OData/Batch/UnbufferedODataBatchHandlerTest.cs

```diff
@@ -168,6 +168,81 @@ public void ProcessBatchAsync_DisposesResponseInCaseOfException()
+                {
+                    content = request.Content.ReadAsStringAsync().Result;
+                    if (!String.IsNullOrEmpty(content))
+                    {
+                        responseContent += "," + content;
+                    }
+                }
+                HttpResponseMessage responseMessage = new HttpResponseMessage { Content = new StringContent(responseContent) };
+                if (content.Equals("foo"))
+                {
+                    responseMessage.StatusCode = HttpStatusCode.BadRequest;
+                }
+                return responseMessage;
+            });
+            UnbufferedODataBatchHandler batchHandler = new UnbufferedODataBatchHandler(server);
+            HttpRequestMessage batchRequest = new HttpRequestMessage(HttpMethod.Post, "http://example.com/$batch")
+            {
+                Content = new MultipartContent("mixed")
+                {
+                    ODataBatchRequestHelper.CreateODataRequestContent(new HttpRequestMessage(HttpMethod.Get, "http://example.com/")),
+                    new MultipartContent("mixed") // ChangeSet
+                    {
+                        ODataBatchRequestHelper.CreateODataRequestContent(new HttpRequestMessage(HttpMethod.Post, "http://example.com/values")
+                        {
+                            Content = new StringContent("foo")
+                        })
+                    },
+                    ODataBatchRequestHelper.CreateODataRequestContent(new HttpRequestMessage(HttpMethod.Post, "http://example.com/values")
+                    {
+                        Content = new StringContent("bar")
+                    }),
+                }
+            };
+            HttpRequestMessage batchRequestWithPrefContinueOnError = new HttpRequestMessage(HttpMethod.Post, "http://example.com/$batch")
+            {
+                Content = new MultipartContent("mixed")
+                {
+                    ODataBatchRequestHelper.CreateODataRequestContent(new HttpRequestMessage(HttpMethod.Get, "http://example.com/")),
+                    new MultipartContent("mixed") // ChangeSet
+                    {
+                        ODataBatchRequestHelper.CreateODataRequestContent(new HttpRequestMessage(HttpMethod.Post, "http://example.com/values")
+                        {
+                            Content = new StringContent("foo")
+                        })
+                    },
+                    ODataBatchRequestHelper.CreateODataRequestContent(new HttpRequestMessage(HttpMethod.Post, "http://example.com/values")
+                    {
+                        Content = new StringContent("bar")
+                    }),
+                }
+            };
+            batchRequestWithPrefContinueOnError.Headers.Add("prefer", "odata.continue-on-error");
+
+            // Act
+            var response = batchHandler.ProcessBatchAsync(batchRequest, CancellationToken.None).Result;
```

Comment:
Check the batch response has a `Preference-Applied` header with values include `odata.continue-on-error`.